osu.Game/Beatmaps/Formats/LegacyBeatmapEncoder.cs

```diff
@@ -233,9 +233,9 @@ private void handleHitObject(TextWriter writer, HitObject hitObject)
             writer.Write(FormattableString.Invariant($"{(int)getObjectType(hitObject)},"));
             writer.Write(FormattableString.Invariant($"{(int)toLegacyHitSoundType(hitObject.Samples)},"));
 
-            if (hitObject is IHasCurve curveData)
+            if (hitObject is IHasPathWithRepeats curveData)
```

Comment:
Tests seem to pass when this is changed to

```suggestion
            if (hitObject is IHasPath curveData)
```

As noted by @smoogipoo drumrolls weren't entering this code path. This seems like an accidental mistype more than anything, seeing that `addPathData` already accepts an `IHasPath` instance.